crates/router/src/db/refund.rs

```diff
@@ -719,23 +745,39 @@ impl RefundInterface for MockDb {
 
     async fn find_refund_by_payment_id_merchant_id(
         &self,
-        _payment_id: &str,
-        _merchant_id: &str,
+        payment_id: &str,
+        merchant_id: &str,
         _storage_scheme: enums::MerchantStorageScheme,
     ) -> CustomResult<Vec<storage_types::Refund>, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let refunds = self.refunds.lock().await;
+
+        Ok(refunds
+            .iter()
+            .take_while(|refund| {
+                refund.merchant_id == merchant_id && refund.payment_id == payment_id
+            })
+            .cloned()
+            .collect::<Vec<_>>())
     }
 
     #[cfg(feature = "olap")]
     async fn filter_refund_by_constraints(
         &self,
-        _merchant_id: &str,
-        _refund_details: &api_models::refunds::RefundListRequest,
+        merchant_id: &str,
+        refund_details: &api_models::refunds::RefundListRequest,
         _storage_scheme: enums::MerchantStorageScheme,
-        _limit: i64,
+        limit: i64,
     ) -> CustomResult<Vec<storage_models::refund::Refund>, errors::StorageError> {
-        // [#172]: Implement function for `MockDb`
-        Err(errors::StorageError::MockDbError)?
+        let refunds = self.refunds.lock().await;
+
+        Ok(refunds
+            .iter()
+            .take_while(|refund| {
+                refund.merchant_id == merchant_id
+                    && refund_details.limit.is_some()
+                    && refund_details.limit == Some(limit)
+            })
```

Comment:
same here, use `filter()` instead. Also use the [take()](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.take) method to apply the limit.

Reply:
`take` has an argument of type `usize` whileas `limit` is of type `i64`
what would you suggest ?